include/term.h

```diff
@@ -72,6 +73,13 @@ class TermIterBase
 class TermIter
 {
  public:
+  // typedefs for marking as an input iterator
+  typedef Term value_type;
+  typedef std::ptrdiff_t difference_type;
+  typedef Term * pointer;
```

Comment:
A bonus comment here for what pointer and reference are doing would be nice.

Reply:
Good idea, thanks! Gave a link to iterator_traits and what it's used for in 70f04dda050865c2182eaeca6d22ad60013d4f2e. Didn't really explain it in detail, but I think it makes sense to accept it as c++ magic

Reply:
It makes more sense now! I hadn't read the inputiterator docs.